src/app/static/src/app/store/projects/actions.ts

```diff
@@ -79,6 +81,38 @@ export const actions: ActionTree<ProjectsState, RootState> & ProjectsActions = {
                     dispatch("load/loadFromVersion", response.data, {root: true})
                 }
             });
+    },
+
+    async deleteProject(context, projectId) {
+        const {commit, dispatch, state} = context;
+
+        if (state.currentProject && state.currentProject.id === projectId) {
+            commit({type: ProjectsMutations.ClearCurrentVersion});
+        }
+        await api<ProjectsMutations, ProjectsMutations>(context)
+            .ignoreSuccess()
+            .withError(ProjectsMutations.ProjectError)
+            .delete(`project/${projectId}`)
+            .then(() => {
+                dispatch("getProjects");
```

Comment:
It seemed easiest here just to refresh projects from the backend following a delete.